junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/ParameterResolutionUtilsTests.java

```diff
@@ -53,38 +50,40 @@
 
 	private final JupiterConfiguration configuration = mock(JupiterConfiguration.class);
 
-	private MutableExtensionRegistry extensionRegistry = MutableExtensionRegistry.createRegistryWithDefaultExtensions(
+	private final MutableExtensionRegistry extensionRegistry = MutableExtensionRegistry.createRegistryWithDefaultExtensions(
 		configuration);
 
 	@Test
-	void constructorInjection() {
-		register(new StringParameterResolver(), new NumberParameterResolver());
+	void resolveConstructorArguments() {
+		register(new StringParameterResolver());
 
-		Class<ConstructorInjectionTestCase> outerClass = ConstructorInjectionTestCase.class;
-		Constructor<ConstructorInjectionTestCase> constructor = ReflectionUtils.getDeclaredConstructor(outerClass);
-		ConstructorInjectionTestCase outer = newInvoker().invoke(constructor, Optional.empty(), extensionContext,
-			extensionRegistry, passthroughInterceptor());
+		Class<ConstructorInjectionTestCase> topLevelClass = ConstructorInjectionTestCase.class;
+		Object[] arguments = resolveConstructorParameters(topLevelClass, null);
+
+		assertThat(arguments).hasSize(1).containsExactly(ENIGMA);
```

Comment:
```suggestion
		assertThat(arguments).containsExactly(ENIGMA);
```

`hasSize()` is already covered by `containsExactly()`